### PR TITLE
feat: text2text and token classification for argilla

### PR DESCRIPTION
**Pull request by MthwRobinson**
### Summary

Closes Unstructured-IO/community#27 and Unstructured-IO/community#28. Adds support for `text2text` and `token_classification` tasks to the `stage_for_argilla` brick. For token classification tasks, if `tokens` is not specified as an optional kwarg, then the `nltk` word tokenizer is used by default.

#### Token Classification

```python
  import json

  from unstructured.documents.elements import Title, NarrativeText
  from unstructured.staging.argilla import stage_for_argilla

  elements = [Title(text="Title"), NarrativeText(text="Narrative")]
  metadata = [{"type": "title"}, {"type": "text"}]

  argilla_dataset = stage_for_argilla(elements, "token_classification", metadata=metadata)
```

#### Text2Text

```python
  import json

  from unstructured.documents.elements import Title, NarrativeText
  from unstructured.staging.argilla import stage_for_argilla

  elements = [Title(text="Title"), NarrativeText(text="Narrative")]
  metadata = [{"type": "title"}, {"type": "text"}]

  argilla_dataset = stage_for_argilla(elements, "text2text", metadata=metadata)
```

### CODE REVIEW

File: docs/source/bricks.rst [hunk 1]
```diff
@@ -901,8 +901,11 @@ Example:
 --------------------------
 
 Convert a list of ``Text`` elements to an `Argilla Dataset <https://docs.argilla.io/en/latest/reference/python/python_client.html#python-ref-datasets>`_.
-The type of Argilla dataset to be generated can be specified with `argilla_task` parameter. Currently, only ``text_classification``
-task type is supported.
+The type of Argilla dataset to be generated can be specified with ``argilla_task``
+parameter. Valid values for ``argilla_task`` are ``"text_classification"``,
+``"token_classification"``, and ``"text2text"``. If ``"token_classification"`` is selected
+and ``tokens`` is not included in the optional recrod kwargs, the ``nltk`` word tokenizer
```
Comment (qued) [inline docs/source/bricks.rst:907]:
```suggestion
and ``tokens`` is not included in the optional kwargs, the ``nltk`` word tokenizer
```
I think this is a typo of record, but I'm not sure what's meant by "optional record kwargs"? I'm suggesting just removing it but that may not capture the intent here.

Comment (MthwRobinson) [reply docs/source/bricks.rst:907]: Thanks, yeah I think it looks fine with 💿 removed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
-## 0.3.0-dev5
+## 0.3.0-dev6
 
-* Implement staging brick for Argilla.
+* Implement staging brick for Argilla. Converts `Text` elements to `argilla` dataset classes.
```
Comment (qued) [inline CHANGELOG.md:3]:
```suggestion
* Implement staging brick for Argilla. Converts lists of `Text` elements to `argilla` dataset classes.
```

**Review by qued — APPROVED** [review]
Other than comments, LGTM!

**Pull request merged**